docs/component-guides/profiles-irsa.md

```diff
@@ -0,0 +1,118 @@
+# IAM Role Service Accounts (IRSA) support for Kubeflow Profiles
+
+# Background
+
+In AWS, [IRSA](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) is a permissions management tool that allows applying AWS IAM permissions boundaries to Kubernetes service accounts (SA). This is useful, for example, to grant one SA permissions to upload test results to a certain bucket in S3 while granting another SA permissions to read from that bucket without modifying the contents. Because the IAM permissions apply to the SA, the permissions boundary is limited to the pod level rather than the nodegroup level. 
+
+In Kubeflow, IRSA is used to provide an isolation boundary at the profile level, allowing admins to scope profiles to their necessary IAM permissions to meet their AWS usage requirements. This is done by making the profile SA an IRSA and by creating profiles with the `AwsIamForServiceAccount` plugin.
```

Comment:
There are few confusing things in this paragraph like:
- `IRSA is used to provide an isolation boundary at the profile level` -> IRSA is used to provide permissions at pod level using the service account. I think this is happening because we are using the term IRSA for 2 things(the feature in this guide itself and the concept from EKS in general)
- making the profile SA an IRSA -> IRSA is a way of doing something it is not a type of resource in K8s or IAM. the more accurate description is: annotating the `default-editor` service account in the namespace created for the profile with the IAM Role specified in the plugin

We can add a few more details to this section. Few suggestions which come to top of my mind:
- Start with what is a profile and explain its used to provide namespace level isolation based on K8s RBAC and Istio RBAC and authorizationPolicy
- Explain what is plugin in profile, say like it can be used to manage resources outside of K8s cluster based on environment
- Then talk about IAMForServiceAccount plugin for profiles which can be used on AWS to provide permissions to pods in using the `default-editor` service account in profile namespace which can be used to access AWS services using an IAM role. Here you can mention it can be used by admins to scope permissions per profile
- Explain that this plugin annotates the `default-editor` service account in the profile namespace with IAM role provided in plugin and adds permissions in the IAM Role for this service account in the profile namespace 
- `default-editor` is used by various services in Kubeflow to launch resources in users namespace and hence the pods inherit the permissions of this service account. Add a note here saying not all services do this by default

Lets see how to use the `IAMForServiceAccount` plugin to provide AWS service permissions to a notebook server instance in the profile namespace

Reply:
made changes based on this